model/inline/src/main/scala/aqua/model/inline/ArrowInliner.scala

```diff
@@ -190,6 +220,77 @@ object ArrowInliner extends Logging {
         .renameExports(streamsToRename)
   }
 
+  private def renameAndResolveAbilities[S: Mangler: Arrows: Exports](
+    name: String,
+    vm: VarModel,
+    t: AbilityType,
+    oldExports: Map[String, ValueModel],
+    oldArrows: Map[String, FuncArrow]
+  ): State[S, (Map[String, String], Map[String, ValueModel], Map[String, FuncArrow])] = {
+    for {
+      newName <- Mangler[S].findNewName(name)
+      newFieldsName = t.fields.mapBoth { case (n, t) =>
+        s"$name.$n" -> s"$newName.$n"
+      }
+      allNewNames = newFieldsName.add((name, newName)).toSortedMap
+    } yield {
+      val (allVars, allArrows) =
+        getVarsAndArrowsFromAbilities(vm.name, Option(newName), t, oldExports, oldArrows)
+      (allNewNames, allVars, allArrows)
+    }
+  }
+
+  private def getVarsAndArrowsFromAbilities(
+    topOldName: String,
+    topNewName: Option[String],
+    st: AbilityType,
+    oldExports: Map[String, ValueModel],
+    oldArrows: Map[String, FuncArrow],
+    valAcc: Map[String, ValueModel] = Map.empty,
+    arrowsAcc: Map[String, FuncArrow] = Map.empty
```

Comment:
Can you provide comments to this function?
Also comments inside body would be great because I don't quite understand what is going on there.